File: distros/L/LLEval/example/fizzbuzz.c
#!lleval
#include <string.h>
#include <stdio.h>
char *fizz(int n){ return n % 3 ? "" : "Fizz"; }
char *buzz(int n){ return n % 5 ? "" : "Buzz"; }
char *fizzbuzz(char *buf, int n){
  strcpy(buf, fizz(n));
  strcat(buf, buzz(n));
  if (!strlen(buf)) sprintf(buf, "%d", n);
  return buf;
}
void main(){
  int i;
  char buf[12];
  for (i = 1; i <= 30; i++){
    printf("%s\n", fizzbuzz(buf, i));
  }
}

File: distros/L/LLEval/example/fizzbuzz.ml
#!lleval
let fizzbuzz n =
  let fizz = if n mod 3 == 0 then "Fizz" else "" in
  let buzz = if n mod 5 == 0 then "Buzz" else "" in
  if [] != List.filter (fun s -> String.length s != 0) [fizz;buzz]
     then fizz ^ buzz
     else string_of_int n

let puts s = print_string( s ^ "\n");;

for i = 1 to 30 do
  puts (fizzbuzz i)
done;;

